src/python/pants/backend/python/target_types.py

```diff
@@ -144,26 +144,46 @@ class ResolvedPexEntryPoint:
 
 @dataclass(frozen=True)
 class ResolvePexEntryPointRequest:
-    """Determine the entry_point by using the `PexEntryPointField` and falling back to the sources
-    field."""
+    """Determine the entry_point by looking at both `PexEntryPointField` and and the sources field.
+
+    In order of precedence, this can be calculated by:
+
+    1. The `entry_point` field having a well-formed value.
+    2. The `entry_point` using a shorthand `:my_func`, and the `sources` field being set. We
+        combine these into `path.to.module:my_func`.
+    3. The `entry_point` being left off, but `sources` defined. We will use `path.to.module`.
+    """
 
     entry_point_field: PexEntryPointField
     sources: PexBinarySources
 
 
 @rule
 async def resolve_pex_entry_point(request: ResolvePexEntryPointRequest) -> ResolvedPexEntryPoint:
-    if request.entry_point_field.value:
-        return ResolvedPexEntryPoint(request.entry_point_field.value)
+    entry_point_value = request.entry_point_field.value
+    if entry_point_value and not entry_point_value.startswith(":"):
+        return ResolvedPexEntryPoint(entry_point_value)
     binary_source_paths = await Get(
         Paths, PathGlobs, request.sources.path_globs(FilesNotFoundBehavior.error)
     )
     if len(binary_source_paths.files) != 1:
-        raise InvalidFieldException(
-            "No `entry_point` was set for the target "
-            f"{repr(request.sources.address)}, so it must have exactly one source, but it has "
-            f"{len(binary_source_paths.files)}."
-        )
+        instructions_url = "https://www.pantsbuild.org/v2.1/docs/python-package-goal#creating-a-pex-file-from-a-pex_binary-target"
```

Comment:
I don't think we want to bake "v2.1" into this URL, so let's just use https://www.pantsbuild.org/docs/python-package-goal#creating-a-pex-file-from-a-pex_binary-target.

Reply:
Stu proposed baking it in the other day: https://pantsbuild.slack.com/archives/C046T6T9U/p1603735964018700?thread_ts=1602610244.363000&cid=C046T6T9U

> while it’s true that linking to v2.0 rather than v2.1 might get you older docs, it also won’t get you a broken link for removed/renamed features/pages
> more likely to be correct on average i think.

Reply:
Also, we don't want to rewrite the docs for 2.0 - this feature will not be backported. We don't want to change the "default" version until 2.1.0 has landed.

Reply:
The trouble is, v2.1 docs will disappear eventually. We only have 5 versions back. 

Probably we need an error URLs system, but for now hardcoding v2.1 is not a great idea.